app/src/main/java/org/oppia/app/topic/review/TopicReviewViewModel.kt

```diff
@@ -1,7 +1,65 @@
 package org.oppia.app.topic.review
 
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
 import androidx.lifecycle.ViewModel
-import org.oppia.app.viewmodel.ObservableViewModel
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.ProfileId
+import org.oppia.app.model.Topic
+import org.oppia.app.topic.review.reviewitemviewmodel.TopicReviewItemViewModel
+import org.oppia.domain.topic.TopicController
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
 
-/** [ViewModel] for showing a list of subtopics. */
-abstract class TopicReviewViewModel : ObservableViewModel()
+/** [ViewModel] for [TopicReviewFragment]. */
+@FragmentScope
+class TopicReviewViewModel @Inject constructor(
+  private val topicController: TopicController,
```

Comment:
Add this here to get the activity which will be sued for your listener/selector.

`private val activity: AppCompatActivity`

Reply:
@rt4914 I should inject the Fragment and not the activity, right?

Reply:
No, inject activity only because the listener is in activity.

Reply:
@PrarabdhGarg Mostly you face a lot of conflicts, almost like redoing the work as there is one PR which might get merge in few minutes which mainly change the entire name of `Review `to `Revision` in the project.

Reply:
https://github.com/oppia/oppia-android/pull/844

Reply:
Also, I was curious, why was the previous method incorrect?

Reply:
It was incorrect because we were unnecessarily using the `Presenter`. 
Also, as mentioned before, after recent PR merge there will be a lot merge conflicts in your PR which you will need to resolve. Take latest develop code in this branch and try to resolve merge conflicts. Mainly you will need to change `Review` word to `Revision`